python/xoscar/collective/core.py

```diff
@@ -236,11 +404,44 @@ async def reduce(
     root: Optional[int] = 0,
     tag: Optional[int] = 0,
     group_name: str = "default",
+    **kwargs,
```

Comment:
Here every collective interface contains ``stream`` option in ``kwargs``, so just add stream rather than kwargs is fine.